pkg/cache/clusterqueue.go

```diff
@@ -51,11 +53,13 @@ type ClusterQueue struct {
 type Cohort struct {
 	Name    string
 	Members sets.Set[*ClusterQueue]
+	// This field will only be set in snapshot. This field equal to the maximum
+	// allocatable generation among its members.
+	AllocatableResourceGeneration int64
```

Comment:
The problem with this approach is that if one of the non maximum gen is changing, the event will not be visible at cohort level.  Summing the might work better, Also making sure that the Gen of a queue starts from 1 will change the cohort Gen when new queues are added to it. 

Since the way the value is computed is described , it makes sense to keep this in the `These fields are only populated for a snapshot` section.

Reply:
Done. Because cache will call `updateResourceGroups`, Gen of a queue starts from 1